processor/src/decoder/mod.rs

```diff
@@ -810,3 +826,47 @@ pub fn build_op_group(ops: &[Operation]) -> Felt {
     assert!(i <= super::OP_GROUP_SIZE, "too many ops");
     Felt::new(group)
 }
+
+// DEBUG INFO
+// ================================================================================================
+
+pub struct DebugInfo {
+    in_debug_mode: bool,
+    operations: Vec<Operation>,
+    assembly_ops: Vec<(usize, AssemblyOp)>,
+}
+
+impl DebugInfo {
+    pub fn new(in_debug_mode: bool) -> Self {
+        Self {
+            in_debug_mode,
+            operations: Vec::<Operation>::new(),
+            assembly_ops: Vec::<(usize, AssemblyOp)>::new(),
+        }
+    }
+
+    /// Returns whether this decoder instance is instantiated in debug mode.
+    pub fn in_debug_mode(&self) -> bool {
+        self.in_debug_mode
+    }
+
+    /// Returns an operation to be executed at the specified clock cycle. Only applicable in debug mode.
+    pub fn get_operation_at(&self, clk: usize) -> Operation {
+        self.operations[clk]
+    }
```

Comment:
nit: i wonder if it would be more consistent to have `pub fn operations(&self) -> &[Operation]` here instead.